Documentation fixes and improvements (#102)
TUM-PIK-ESM/Terrarium.jl: docs/src/processes/soil/soil_energy.md

```math
\begin{equation}
    \label{eq:energyconservation}
\frac{\partial U(T,\theta)}{\partial t} = - \boldsymbol{\nabla} \cdot \left(\mathbf{j}_\text{h} + \mathbf{j}_h^{\text{w}}\right) + F_h(z,t),
\end{equation}
```
where $U(T,\theta)$ (J/m³) is the volumetric internal energy as a function of temperature and total water/ice content $\theta$ (m³/m³), and $F_h(z,t)$ is an inhomogeneous heat source/sink (forcing) term.

The advective heat flux $\mathbf{j}_{\text{h}}^{\text{w}}$ can be represented as,
```math
\begin{equation}
\mathbf{j}_{\text{h}}^{\text{w}} = \left( c_{\text{w}} T + L_{\text{sl}} \right) \mathbf{j}_{\text{w}} \rho_{\text{w}}
\end{equation}
```
where $L_{\text{sl}}$ (J/kg) and $c_{\text{w}}$ (J/(kg K)) represent the specific latent heat of fusion (solid → liquid) and heat capacity of liquid water respectively. Note that $\mathbf{j}_{\mathrm{w}}$ is the water flux vector as defined in [Soil hydrology](@ref). This flux term accounts for the energy transferred by the movement of water within the soil matrix. In model configurations that neglect subsurface water flow, this flux term is implicitly assumed to be zero.

!!! todo "Advective heat flux"

>     The current implementation does not yet consider the advective heat flux, but this is work in progress! 

Reviewer: typo, this is *a* work in progress

>     The current implementation does not yet consider the advective heat flux, but this is work in progress! 

Reviewer: ```suggestion
    The current implementation does not yet consider the advective heat flux, but this will be added soon!
```
